optuna/pruners/hyperband.py

```diff
@@ -94,25 +98,28 @@ class HyperbandPruner(BasePruner):
     def __init__(
         self,
         min_resource: int = 1,
-        max_resource: int = 80,
+        max_resource: Union[str, int] = "auto",
         reduction_factor: int = 3,
         n_brackets: Optional[int] = None,
         min_early_stopping_rate_low: Optional[int] = None,
     ) -> None:
 
-        self._pruners = []  # type: List[SuccessiveHalvingPruner]
+        self._min_resource = min_resource
+        self._max_resource = max_resource
         self._reduction_factor = reduction_factor
+        self._n_brackets = n_brackets
+        self._min_early_stopping_rate_low = min_early_stopping_rate_low
+        self._pruners = []  # type: List[SuccessiveHalvingPruner]
         self._total_trial_allocation_budget = 0
+        self._trial_allocation_budgets = []  # type: List[int]
 
-        if n_brackets is None:
-            # In the original paper http://www.jmlr.org/papers/volume18/16-558/16-558.pdf, the
-            # inputs of Hyperband are `R`: max resource and `\eta`: reduction factor. The
-            # number of brackets (this is referred as `s_{max} + 1` in the paper) is calculated
-            # by s_{max} + 1 = \floor{\log_{\eta} (R)} + 1 in Algorithm 1 of the original paper.
-            self._n_brackets = (
-                math.floor(math.log(max_resource / min_resource, reduction_factor)) + 1
+        if not isinstance(self._max_resource, int) and self._max_resource != "auto":
+            raise ValueError(
+                "The 'max_resource' should be integer or 'auto'. "
+                "But max_resource = {}".format(self._max_resource)
```

Comment:
Is there any test on this exception?

Reply:
There is no test on this exception. I will add it. Thank you for your pointing!!